examples/examples_yaml_test.go

```diff
@@ -208,6 +208,43 @@ func TestYamlDeploymentSettingsExample(t *testing.T) {
 	})
 }
 
+func TestYamlDeploymentSettingsNoSourceExample(t *testing.T) {
+
+	// Set up tmpdir with a Pulumi.yml with no resources
```

Comment:
How does this work? Is the ProgramTest running `up` on the folder with Pulumi.yml and then on a non-existing folder? Why is this needed?

Reply:
This is copy-paste from the other TestYamlDeploymentSettingsExample above.

Reply:
Duh, should've scrolled

Still curious as to why we need these temp dirs, but I can take a look later, unrelated to the change